app/routes.py

```diff
@@ -1 +1,239 @@
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = validate_model(Task,task_id)
+
+    task.completed_at= None
+    
+    db.session.commit()
+    
+    return task.to_dic(), 200
+
+
+# DELETE TASK- DELETE request to /tasks/<task_id>
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_book(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return { "details":f"Task {task.task_id} \"{task.title}\" successfully deleted"}, 200
+
+
+
+# CREATE GOAL- POST request to /goals
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    
+    if "title" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    
+    new_goal = Goal(title=request_body["title"])
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": {
+        "id": new_goal.goal_id,
+        "title": new_goal.title,
+    }}, 201  
+
+
+
+# READ ALL GOALS- GET request to /goals
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+    goal_sort_query = request.args.get("sort")
+
+    if goal_sort_query:
+        goals = Goal.query.order_by(Goal.title).all()
+    else:
+        goals = Goal.query.all()
+        
+    goals_response = []
+    
+    for goal in goals:
+        goals_response.append(goal.to_dic()["goal"])
+        
+    return jsonify(goals_response), 200
+
+# READ ONE GOAL- GET request to /goals/<goal_id>
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return goal.to_dic(), 200
+    
+    
+# UPDATE GOAL- PATCH request to /goals/<goal_id>/mark_incomplete
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    title = request_body["title"]
+
+    db.session.commit()
+
+    return goal.to_dic()
+
+# DELETE GOAL- DELETE request to /goals/<goal_id>
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({'details': f'Goal {goal_id} \"{goal.title}\" successfully deleted'}, 200)
+
+@goals_bp.route("<goal_id>/tasks", methods=["POST"])
+def add_tasks_to_goal_by_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    for id in request_body["task_ids"]:
+        task = validate_model(Task, id)
+        goal.tasks.append(task)
+    
+    db.session.commit()
+    
+    return {
+        "id": goal.goal_id,
+        "task_ids": request_body["task_ids"]
+    }
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_task_by_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    tasks_response = goal.task_by_goal_id()
+    
+    return jsonify(tasks_response)
+    
+    
+
```

Comment:
Well done on this project, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭